### ISSUE TITLE
[UX][Design] Research and Recommend Ways Forward for this Issue

### ISSUE THREAD

**Issue by sarahknoppA6** [department-of-veterans-affairs/va.gov-team#45956]
## Background
There were multiple ways identified that a patient can check into an appointment they've already checked into: https://dsva.slack.com/archives/C022AC2STBM/p1659368405831289?thread_ts=1659364651.031189&cid=C022AC2STBM

## Tasks
- Research the scenarios in the slack thread above
- Recommend user workflows for the scenarios

**Comment by kristenmcconnell** [comment]
Initial questions and thoughts

**Scenario 1:  Patient already checked themselves in**
- This looks to be corrected, but the statement returned to the patient is assuming the patient checked themselves in which isn't validated.  
 
- Do we want to change wording to show that staff or Veterans do the checking in?
    - Maybe talk to Danielle about this.
    - This would impact the SMS and appointment list messages. Anything else? 

**Scenario 2:  Appointment is checked in before receiving check-in URL** - This still exists.  Appointment is checked in by staff, patient texts "check in", is returned check-in URL, patient logs in and is let through the majority of the workflow and only after getting to the appointments page do we tell them they can't proceed because they're already checked in.  This should have stopped the patient at the text message layer to redirect them to staff if there's no appointment that can be checked in.  

- There is already supposed to be a text message to capture this scenario if the Veteran was just starting the workflow. Does it not exist? [User flow screenshot]
- Is this occurring due to the time it takes the staff side to notify the Veteran-side? Can anything be done about this? 

**Scenario 3:  Appointment is checked in after receiving check-in URL** - This still exists.  Appointment is not checked in, patient texts "check in", is returned check-in URL, staff check appointment in, patient logs in and is let through the majority of the workflow and only after getting to the appointments page do we tell them they can't proceed because they're already checked in.  This should have stopped them after logging in.

- How often is this happening? Can we tell? Though how much work is it to add another error state?
- Option 1: Add error message right after log in if the Veteran checked in with staff, but already had a check-in URL that was still valid.	
 - Option 2: Do nothing. Let them review the appointment list.

Additionally, the verbiage is not encompassing of all scenarios.  There's no logic check to see who checked the appointment in, it's assuming the patient checked themselves in by presenting You checked in at <time>.  A more appropriate and encompassing statement would be You were checked in at <time> as the current statement is incorrectly stating the patient checked themselves in for scenarios 2 and 3.

- Is the wording in our app all inclusive of staff checked in appointments? Do Veterans care about the difference between who checked them in (themself or staff)?

**Comment by kristenmcconnell** [comment]
@Hortonio I wanted to run something by you...

For Scenario 2 listed in the above comment, can you confirm that the SMS messages in the image are in production?

**Comment by Hortonio** [comment]
@kristenmcconnell Sure thing!

Scenario 2: Check-in too early or too late

Straight from the source code for a Veteran trying to check-in too early, the following message is live in prod: 

const CHECK_IN_EARLY = (MINUTES_BEFORE) => `Our records show that you have an appointment today. You can check in up to ${MINUTES_BEFORE} minutes before your appointment. Try again later.`;

For a Veteran trying to check-in too late, the following message was updated and live in prod per ticket [937](https://app.zenhub.com/workspaces/chip-api-6149f6401b0b3c001219802f/issues/department-of-veterans-affairs/chip/937):

const CHECK_IN_LATE = `We\'re sorry. Online check-in is no longer available for your appointment time. Ask a staff member for help.`;

**Comment by kristenmcconnell** [comment]
@Hortonio One more question...and CHIP is checking that the appointments aren't already checked into before the URL is sent? I'm just trying to figure out how Shawn saw what he saw, because I thought that it shouldn't be possible with what we already developed.

**Comment by Hortonio** [comment]
@kristenmcconnell Correct; Architecture-wise, CHIP should be performing a check of appointment check-in status before sending URL.

**Comment by kristenmcconnell** [comment]
Thanks, Chris. It sounds like Scenario 2 might be a bug that Shawn caught. Can we write a ticket for engineering to look into it more and talk to Shawn about how exactly he found it? @loripusey @Hortonio  I don't think there is any further UX work to be done around Scenario 2. The user workflows and content work that we have already designed should account for it.

**Comment by Hortonio** [comment]
Perhaps a bug indeed. Logged bug ticket [1216](https://app.zenhub.com/workspaces/chip-api-6149f6401b0b3c001219802f/issues/department-of-veterans-affairs/chip/1216), just need the specific example to reference for research.

**Comment by loripusey** [comment]
Scenario 3 seems like an edge case. Not sure how many would try to check in after checking-in with an MSA.

**Comment by kristenmcconnell** [comment]
@loripusey I agree that Scenario 3 seems like an edge case. I was also thinking that it might be best to just let the Veteran through to the appointment list page if Scenario 3 does arise, so they could see exact status for each appointment. Therefore, there would be no change to any workflow. 

I am going to check with Danielle on Thursday about our wording and if we can make some small changes to imply both Veteran and staff check-ins. However, I think that our wording is pretty generic. I'll see what she thinks. This would address Scenario 1.

Once I checked with Danielle, then I believe the scope of this ticket has been addressed.

However, this ticket make me think about our error states when a Veteran has multiple appointments. We only have SMS messages that cover:

- Appointments are all after the check in window (appoinmentsService)
  - SMS Wording: We're sorry. Online check-in is no longer available for your appointment time. Ask a staff member for help.
- Appointments are all before the check in window (appointmentService)
  - SMS Wording: Our records show that you have an appointment today. You can check in up to ${process.env.

Do we need additional SMS error messages that better cover when a Veteran has some appointments before and after the check in window? Should the SMS message add checked in status information? Is there a HIPAA reason not to do that? Should we created SMS messages for the following scenarios:

- All appointments before the check in window have been already checked in, but future appointment(s) still exists that day
- All appointments before the check in window have been already checked in, but no future appointment(s) for that day exist
- All appointments before the check in window have not been checked in, but future appointment(s) still exist that day
- All appointment before the check in window have not been checked in, but no future appointment(s) for that day exist
- Some appointment before the check in window have been checked in and some haven't, but future appointment(s) still exist that day
- Some appointment before the check in window have been checked in and some haven't, but no future appointment(s) for that day exist

This is probably a separate ticket. We could discuss if we think this is a good idea or not at Friday's UX Sync.

**Comment by kristenmcconnell** [comment]
Met with Danielle. The wording "You checked in at [time]." is pretty agnostic with regard to check-in method (with staff or PCI). 

@benbrasso-agile6  talked with Shawn Adams, who brought up his concern that people might be missed in the waiting room. We believe that might be the reasoning behind the interest in adding whether someone checked in via PCI or staff. We think there might be another way to do that, so Ben is adding a ticket to the backlog.
